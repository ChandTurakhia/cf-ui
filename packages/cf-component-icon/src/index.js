import IconUnstyled from './Icon';
import IconTheme from './IconTheme';

import { applyTheme } from 'cf-style-container';

const Icon = applyTheme(IconUnstyled, IconTheme);

export { Icon, IconUnstyled, IconTheme };
