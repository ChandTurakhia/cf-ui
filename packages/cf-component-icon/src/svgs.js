import api from './svgs/api.svg';
import bolt from './svgs/bolt.svg';
import calendar from './svgs/calendar.svg';
import caretdown from './svgs/caret-down.svg';
import caretleft from './svgs/caret-left.svg';
import caretright from './svgs/caret-right.svg';
import caretup from './svgs/caret-up.svg';
import chart from './svgs/chart.svg';
import clipboard from './svgs/clipboard.svg';
//TODO add actual close svg [Used once on notifications]
import close from './svgs/clipboard.svg';
import creditcard from './svgs/credit-card.svg';
import door from './svgs/door.svg';
import download from './svgs/download.svg';
import drive from './svgs/drive.svg';
import exclamationsign from './svgs/exclamation-sign.svg';
import facebook from './svgs/facebook.svg';
import file from './svgs/file.svg';
import filter from './svgs/filter.svg';
import firebolt from './svgs/firebolt.svg';
import flowchart from './svgs/flowchart.svg';
import gear from './svgs/gear.svg';
import googleplus from './svgs/google-plus.svg';
import hamburger from './svgs/hamburger.svg';
import happy from './svgs/happy.svg';
import help from './svgs/help.svg';
import infosign from './svgs/info-sign.svg';
import linkedin from './svgs/linkedin.svg';
import list from './svgs/list.svg';
//TODO add actual loading svg
import loading from './svgs/list.svg';
import lock from './svgs/lock.svg';
import mail from './svgs/mail.svg';
import network from './svgs/network.svg';
import oksign from './svgs/ok-sign.svg';
import ok from './svgs/ok.svg';
import pause from './svgs/pause.svg';
import plus from './svgs/plus.svg';
import refresh from './svgs/refresh.svg';
import remove from './svgs/remove.svg';
import resizehorizontal from './svgs/resize-horizontal.svg';
import sad from './svgs/sad.svg';
import search from './svgs/search.svg';
import shield from './svgs/shield.svg';
import speech from './svgs/speech.svg';
import time from './svgs/time.svg';
import twitter from './svgs/twitter.svg';
import upload from './svgs/upload.svg';
import wrench from './svgs/wrench.svg';

export {
  api,
  bolt,
  calendar,
  caretdown,
  caretleft,
  caretright,
  caretup,
  chart,
  clipboard,
  close,
  creditcard,
  door,
  download,
  drive,
  exclamationsign,
  facebook,
  file,
  filter,
  firebolt,
  flowchart,
  gear,
  googleplus,
  hamburger,
  happy,
  help,
  infosign,
  linkedin,
  list,
  loading,
  lock,
  mail,
  network,
  oksign,
  ok,
  pause,
  plus,
  refresh,
  remove,
  resizehorizontal,
  sad,
  search,
  shield,
  speech,
  time,
  twitter,
  upload,
  wrench
};
