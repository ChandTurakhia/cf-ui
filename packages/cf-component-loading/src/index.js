import LoadingUnstyled from './Loading';
import LoadingTheme from './LoadingTheme';

import { applyTheme } from 'cf-style-container';

const Loading = applyTheme(LoadingUnstyled, LoadingTheme);

export { Loading, LoadingUnstyled, LoadingTheme };
