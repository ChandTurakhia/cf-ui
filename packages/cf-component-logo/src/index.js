import Logo from './Logo';
import logoSvg from '../assets/logo.svg';
import logoSvgMobile from '../assets/logo_mobile.svg';

export { Logo, logoSvg, logoSvgMobile };
