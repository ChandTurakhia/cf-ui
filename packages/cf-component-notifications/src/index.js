import NotificationUnstyled from './Notification';
import NotificationListUnstyled from './NotificationList';
import NotificationGlobalContainerUnstyled from './NotificationGlobalContainer';

import NotificationTheme from './NotificationTheme';
import NotificationListTheme from './NotificationListTheme';
import NotificationGlobalContainerTheme
  from './NotificationGlobalContainerTheme';

import { applyTheme } from 'cf-style-container';

const Notification = applyTheme(NotificationUnstyled, NotificationTheme);

const NotificationList = applyTheme(
  NotificationListUnstyled,
  NotificationListTheme
);

const NotificationGlobalContainer = applyTheme(
  NotificationGlobalContainerUnstyled,
  NotificationGlobalContainerTheme
);

export {
  Notification,
  NotificationList,
  NotificationGlobalContainer,
  NotificationUnstyled,
  NotificationListUnstyled,
  NotificationGlobalContainerUnstyled,
  NotificationTheme,
  NotificationListTheme,
  NotificationGlobalContainerTheme
};
