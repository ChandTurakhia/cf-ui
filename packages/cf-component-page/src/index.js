import PageUnstyled from './Page';
import PageHeaderUnstyled from './PageHeader';
import PageContentUnstyled from './PageContent';
import PageTheme from './PageTheme';
import Header from './Header';
import Logo from './Logo';
import NavList from './NavList';
import NavItem from './NavItem';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

import { applyTheme } from 'cf-style-container';

const Page = applyTheme(PageUnstyled, PageTheme);
const PageHeader = applyTheme(PageHeaderUnstyled, {});
const PageContent = applyTheme(PageContentUnstyled, {});

export {
  Header,
  Logo,
  logoSvg,
  logoSvgMobile,
  NavItem,
  NavList,
  Page,
  PageTheme,
  PageUnstyled,
  PageHeader,
  PageHeaderUnstyled,
  PageContent,
  PageContentUnstyled
};
