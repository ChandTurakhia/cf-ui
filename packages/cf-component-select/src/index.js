import { applyTheme } from 'cf-style-container';

import SelectUnstyled from './Select';
import SelectTheme from './SelectTheme';

const Select = applyTheme(SelectUnstyled, SelectTheme);

export { Select, SelectUnstyled, SelectTheme };
