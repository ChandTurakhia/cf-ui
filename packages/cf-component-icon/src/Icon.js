import React from 'react';
import PropTypes from 'prop-types';

import * as svgs from './svgs.js';

import { createComponent } from 'cf-style-container';

const DEFAULT_HEIGHT = 15.5;

const getHeight = size => {
  let height;
  switch (size) {
    case '1.5x':
      height = 23;
      break;
    case '2x':
      height = 30.5;
      break;
    case '2.5x':
      height = 38;
      break;
    case '3x':
      height = 45.5;
      break;
    case '3.5x':
      height = 53;
      break;
    case '4x':
      height = 60.5;
      break;
    default:
      height = DEFAULT_HEIGHT;
  }
  return height;
};

const getFill = (color, theme) => {
  let fill;
  switch (color) {
    case 'default':
      fill = theme.color.smoke;
      break;
    case 'primary':
      fill = theme.color.marine;
      break;
    case 'success':
      fill = theme.color.grass;
      break;
    case 'warning':
      fill = theme.color.carrot;
      break;
    case 'danger':
      fill = theme.color.apple;
      break;
    case 'black':
      fill = theme.colorBlack;
      break;
    case 'white':
      fill = theme.colorWhite;
      break;
    default:
      fill = theme.colorBlack;
  }
  return fill;
};

const IconWrapper = createComponent(
  ({ height }) => ({
    display: 'inline-block',
    height: height ? height : DEFAULT_HEIGHT
  }),
  'span',
  ['aria-label']
);

class Icon extends React.Component {
  render() {
    const { className, size, label, color } = this.props;
    const icon = svgs[this.props.type.replace(/-/g, '')];
    let height, fill;

    if (!icon) throw `Invalid Icon type ${this.props.type}`;

    if (size) {
      height = getHeight(size);
    }

    const SvgIcon = createComponent(
      ({ color, theme }) => ({
        height: '100%',
        fill: color ? getFill(color, theme) : theme.colorBlack
      }),
      icon
    );

    return (
      <IconWrapper aria-label={label} height={height}>
        <SvgIcon color={color} />
      </IconWrapper>
    );
  }
}

Icon.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.oneOf([false])])
    .isRequired,
  type: PropTypes.oneOf([
    'api',
    'bolt',
    'calendar',
    'caret-down',
    'caret-left',
    'caret-right',
    'caret-up',
    'chart',
    'clipboard',
    'close',
    'credit-card',
    'door',
    'download',
    'drive',
    'exclamation-sign',
    'facebook',
    'file',
    'filter',
    'firebolt',
    'flowchart',
    'gear',
    'googleplus',
    'hamburger',
    'happy',
    'help',
    'info-sign',
    'linkedin',
    'list',
    'loading',
    'lock',
    'mail',
    'network',
    'ok-sign',
    'ok',
    'pause',
    'plus',
    'refresh',
    'remove',
    'resize-horizontal',
    'sad',
    'search',
    'shield',
    'speech',
    'time',
    'twitter',
    'upload',
    'wrench'
  ]).isRequired,
  size: PropTypes.oneOf(['2x', '2.5x', '3x', '3.5x', '4x']),
  color: PropTypes.oneOf([
    'default',
    'primary',
    'success',
    'warning',
    'danger',
    'black',
    'white'
  ])
};

export default Icon;
