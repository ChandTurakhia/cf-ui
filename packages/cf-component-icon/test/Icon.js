import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { Icon } from '../../cf-component-icon/src/index';

test('should render type', () => {
  const snapshot = felaSnapshot(<Icon type="info-sign" label="Info" />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render type/size', () => {
  const snapshot = felaSnapshot(
    <Icon type="hamburger" label="Info" size="2x" />
  );
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render type with no label', () => {
  const snapshot = felaSnapshot(<Icon type="info-sign" label={false} />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render colors', () => {
  const snapshot = felaSnapshot(
    <div>
      <Icon label="default" size="2x" type="search" color="default" />
      <Icon label="primary" size="2x" type="caret-right" color="primary" />
      <Icon label="success" size="2x" type="ok" color="success" />
      <Icon label="warning" size="2x" type="info-sign" color="warning" />
      <Icon label="danger" size="2x" type="exclamation-sign" color="danger" />
      <Icon label="black" size="2x" type="remove" color="black" />
      <Icon label="white" size="2x" type="shield" color="white" />
    </div>
  );
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
