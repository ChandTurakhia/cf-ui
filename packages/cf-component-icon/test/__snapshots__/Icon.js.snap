// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render colors 1`] = `
<div>
  <svg
    aria-label="default"
    className="a b"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.69,13.679l-3.25-3.25a5.791,5.791,0,1,0-1.315,1.17l3.322,3.322a.37.37,0,0,0,.524,0l.719-.718A.371.371,0,0,0,14.69,13.679ZM4.137,9.595a4.026,4.026,0,1,1,5.8-.125l-.24.24a4.014,4.014,0,0,1-5.563-.114Z"
    />
  </svg>
  <svg
    aria-label="primary"
    className="a c"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.537,11.483l3.3-3.3a.263.263,0,0,0,0-.371l-3.3-3.3a.262.262,0,0,0-.448.186v6.6A.262.262,0,0,0,6.537,11.483Z"
    />
  </svg>
  <svg
    aria-label="success"
    className="a d"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="15.736 3.872 13.7 1.836 5.761 9.776 2.3 6.315 0.264 8.35 5.761 13.847 15.736 3.872"
    />
  </svg>
  <svg
    aria-label="warning"
    className="a e"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.149,1.014A6.986,6.986,0,1,0,15.134,8,6.986,6.986,0,0,0,8.149,1.014Zm.817,11.423h-1.7V6.933h1.7Zm0-7.065a.226.226,0,0,1-.226.226H7.487a.226.226,0,0,1-.226-.226V4.121a.226.226,0,0,1,.226-.226H8.739a.226.226,0,0,1,.226.226Z"
    />
  </svg>
  <svg
    aria-label="danger"
    className="a f"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.672,14.2,8.3,1.443a.352.352,0,0,0-.609,0L.328,14.2a.352.352,0,0,0,.3.527H15.368A.352.352,0,0,0,15.672,14.2ZM8.743,12.9a.221.221,0,0,1-.221.221H7.478a.221.221,0,0,1-.221-.221V11.86a.221.221,0,0,1,.221-.221H8.522a.221.221,0,0,1,.221.221Zm-.025-2.422H7.282L7.257,6.005H8.743Z"
    />
  </svg>
  <svg
    aria-label="black"
    className="a g"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="14.3 3.625 12.375 1.7 8 6.075 3.625 1.7 1.7 3.625 6.075 8 1.7 12.375 3.625 14.3 8 9.925 12.375 14.3 14.3 12.375 9.925 8 14.3 3.625"
    />
  </svg>
  <svg
    aria-label="white"
    className="a h"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.98,1.312V8.43h0a5.172,5.172,0,0,0,1.168,3.153,5.921,5.921,0,0,0,1.908,1.54l.01.012L8,14.688l3.008-1.593.016-.013a5.9,5.9,0,0,0,1.825-1.5A5.172,5.172,0,0,0,14.017,8.43h0V1.312Zm10.761,7.03h0a4.086,4.086,0,0,1-.921,2.489,4.668,4.668,0,0,1-1.441,1.183l-.013.01L7.989,13.282V2.724h4.751Z"
    />
  </svg>
</div>
`;

exports[`should render colors 2`] = `
"
.a {
  height: 30px
}

.b {
  fill: #7D7D7D
}

.c {
  fill: #2F7BBF
}

.d {
  fill: #9BCA3E
}

.e {
  fill: #f56500
}

.f {
  fill: #BD2527
}

.g {
  fill: #000
}

.h {
  fill: #fff
}
"
`;

exports[`should render type 1`] = `
<svg
  aria-label="Info"
  className="a b"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M8.149,1.014A6.986,6.986,0,1,0,15.134,8,6.986,6.986,0,0,0,8.149,1.014Zm.817,11.423h-1.7V6.933h1.7Zm0-7.065a.226.226,0,0,1-.226.226H7.487a.226.226,0,0,1-.226-.226V4.121a.226.226,0,0,1,.226-.226H8.739a.226.226,0,0,1,.226.226Z"
  />
</svg>
`;

exports[`should render type 2`] = `
"
.a {
  height: 15px
}

.b {
  fill: #000
}
"
`;

exports[`should render type with no label 1`] = `
<svg
  aria-label=""
  className="a b"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M8.149,1.014A6.986,6.986,0,1,0,15.134,8,6.986,6.986,0,0,0,8.149,1.014Zm.817,11.423h-1.7V6.933h1.7Zm0-7.065a.226.226,0,0,1-.226.226H7.487a.226.226,0,0,1-.226-.226V4.121a.226.226,0,0,1,.226-.226H8.739a.226.226,0,0,1,.226.226Z"
  />
</svg>
`;

exports[`should render type with no label 2`] = `
"
.a {
  height: 15px
}

.b {
  fill: #000
}
"
`;

exports[`should render type/size 1`] = `
<svg
  aria-label="Info"
  className="a b"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.3,1.7V4.22H1.7V1.7ZM1.7,14.3H14.3V11.78H1.7Zm0-5.04H14.3V6.74H1.7Z"
  />
</svg>
`;

exports[`should render type/size 2`] = `
"
.a {
  height: 30px
}

.b {
  fill: #000
}
"
`;
